osu.Game/Overlays/FirstRunSetup/ScreenImportFromStable.cs

```diff
@@ -0,0 +1,268 @@
+
+            allowInteraction(true);
+            stableLocatorTextBox.Current.Value = storage.GetFullPath(string.Empty);
+            importButton.Enabled.Value = true;
+        }
+
+        private void runImport()
+        {
+            allowInteraction(false);
+            progressText.FadeIn(1000, Easing.OutQuint);
+
+            StableContent importableContent = 0;
+
+            foreach (var c in contentCheckboxes.Where(c => c.Current.Value))
+                importableContent |= c.StableContent;
+
+            legacyImportManager.ImportFromStableAsync(importableContent, false).ContinueWith(t => Schedule(() =>
+            {
+                progressText.FadeOut(500, Easing.OutQuint);
+
+                if (t.IsCompletedSuccessfully)
+                    importButton.Complete();
+                else
+                {
+                    allowInteraction(true);
+                    importButton.Abort();
+                }
+            }));
+        }
+
+        private void allowInteraction(bool allow)
+        {
+            importButton.Enabled.Value = allow;
+            stableLocatorTextBox.Current.Disabled = !allow;
+            foreach (var c in contentCheckboxes)
+                c.Current.Disabled = !allow;
+        }
+
+        private class ImportCheckbox : SettingsCheckbox
+        {
+            public readonly StableContent StableContent;
+
+            private readonly LocalisableString title;
+
+            [Resolved]
+            private LegacyImportManager legacyImportManager { get; set; } = null!;
+
+            private CancellationTokenSource? countUpdateCancellation;
+
+            public ImportCheckbox(LocalisableString title, StableContent stableContent)
+            {
+                this.title = title;
+
+                StableContent = stableContent;
+
+                Current.Default = true;
+                Current.Value = true;
+
+                LabelText = title;
+            }
+
+            public void UpdateCount()
+            {
+                LabelText = LocalisableString.Interpolate($"{title} ({FirstRunOverlayImportFromStableScreenStrings.Calculating})");
+
+                countUpdateCancellation?.Cancel();
+                countUpdateCancellation = new CancellationTokenSource();
+
+                legacyImportManager.GetImportCount(StableContent, countUpdateCancellation.Token).ContinueWith(task => Schedule(() =>
+                {
+                    if (task.IsCanceled)
+                        return;
+
+                    int count = task.GetResultSafely();
+
+                    LabelText = LocalisableString.Interpolate($"{title} ({FirstRunOverlayImportFromStableScreenStrings.Items(count)})");
+                }));
+            }
+        }
+
+        internal class StableLocatorLabelledTextBox : LabelledTextBoxWithPopover, ICanAcceptFiles
+        {
+            [Resolved]
+            private LegacyImportManager legacyImportManager { get; set; } = null!;
+
+            // TODO: test
```

Comment:
What's this?

Reply:
Can be removed, was a self-reminder and was subsequently tested 😓